app/data/tasks/deduplicate_records.py

```diff
@@ -0,0 +1,185 @@
+
+
+def get_time_allowance():
+    return datetime.timedelta(hours=settings.DEDUPE_TIME_RANGE_HOURS)
+
+
+# .001 degrees is ~110m give or take 10 depending on location, which should
+# be good enough
+def get_distance_allowance():
+    return settings.DEDUPE_DISTANCE_DEGREES
+
+
+def calculate_score(record, duplicate):
+    """
+    Set a score representing the likelyhood the two records are duplicated, based
+    on the criteria.  score is a float between 0 and 1
+
+    normalized dimensions are considered equal (1 day equally as important as 100m)
+    """
+    tdelta = abs(record.occurred_from - duplicate.occurred_from)
+    tscore = 0
+    if tdelta.total_seconds() == 0:
+        tscore = 1.0
+    else:
+        tscore = tdelta.total_seconds() / get_time_allowance().total_seconds()
+    ddelta = record.geom.distance(duplicate.geom)
+    dscore = 0
+    if ddelta == 0.0:
+        dscore = 1.0
+    else:
+        dscore = ddelta / get_distance_allowance()
+    score = (tscore + dscore) / 2
+    return score
+
+
+def find_duplicates_for_record(uuid):
+    """ Given a record, find all possible duplicates for it
+    """
+    record = Record.objects.get(uuid=uuid)
+    return Record.objects.filter(
+        schema__record_type=record.schema.record_type,
+        geom__dwithin=(record.geom, get_distance_allowance()),
+        occurred_from__range=(
+            record.occurred_from - get_time_allowance(), record.occurred_from + get_time_allowance()
+        )
+    ).exclude(uuid=record.uuid)
+
+
+def get_time_extent():
+    """ Get the time range to dedupe
+    Calculate from X hr before the last dedupe run to the current task run's
+    start time, where X is the heuristic's time allowance
+
+    Returns:
+        dictionary containing the start and end times for the next dedupe task
+        {starttime: timestamp, endtime: timestamp}
+    """
+    # TODO: get config var determining matching time - use value of 1 hr for now
+    time_allowance = get_time_allowance()
+    last_job = None
+    if DedupeJob.objects.all().count() > 0:
+        last_job = DedupeJob.objects.latest()
+    start_time = None
+    if last_job:
+        start_time = last_job.datetime - time_allowance
+    else:
+        start_time = Record.objects.earliest('occurred_from').occurred_from
+    end_time = datetime.datetime.utcnow().replace(tzinfo=timezone('UTC'))
+    return {'start_time': start_time, 'end_time': end_time}
+
+
+def get_dedupe_set(extent):
+    """ Get the set of records to dedupe
+    return a set of uuids to check for duplicates, and the queryset used to
+    generate it
+    """
+    queryset = Record.objects.filter(
+        occurred_from__range=(
+            extent['start_time'], extent['end_time']
+        )
+    ).exclude(uuid__in=get_dedupe_ids())
+    ids = list(queryset.values_list(queryset.model._meta.pk.name, flat=True))
+    return (ids, queryset)
+
+
+def get_dedupe_ids(job=None):
+    """ Get the ids of records to dedupe
```

Comment:
This comment doesn't seem to match what the function does; it looks like it returns a list of ids of records that have ever been flagged as a potential duplicate, and this causes `get_dedupe_set` to ignore those records.